packages/edit-site/src/store/actions.js

```diff
@@ -111,14 +111,39 @@ export function* addTemplate( template ) {
  * @param {Object} template The template object.
  */
 export function* removeTemplate( template ) {
-	yield controls.dispatch(
-		coreStore,
-		'deleteEntityRecord',
-		'postType',
-		template.type,
-		template.id,
-		{ force: true }
-	);
+	try {
+		yield controls.dispatch(
+			coreStore,
+			'deleteEntityRecord',
+			'postType',
+			template.type,
+			template.id,
+			{ force: true }
+		);
+
+		yield controls.dispatch(
+			noticesStore,
+			'createSuccessNotice',
+			sprintf(
+				/* translators: The template/part's name. */
+				__( '"%s" removed.' ),
+				template.title.rendered
+			),
+			{ type: 'snackbar' }
+		);
+	} catch ( error ) {
+		const errorMessage =
+			error.message && error.code !== 'unknown_error'
+				? error.message
+				: __( 'Removing failed.' );
```

Comment:
```suggestion
				: __( 'An error occurred while removing the template.' );
```

Reply:
Let's use "deleting" rather than "removing". It's more explicit and aligns with https://github.com/WordPress/gutenberg/pull/36815